activejob/lib/active_job/arguments.rb

```diff
@@ -14,13 +14,14 @@ def initialize(e) #:nodoc:
     end
   end
 
-  # Raised when an unsupported argument type is being set as job argument. We
-  # currently support NilClass, Fixnum, Float, String, TrueClass, FalseClass,
-  # Bignum and object that can be represented as GlobalIDs (ex: Active Record).
-  # Also raised if you set the key for a Hash something else than a string or
-  # a symbol.
-  class SerializationError < ArgumentError
-  end
+  # Raised when an unsupported argument type is being set as job argument.
+  # We currently support the types in +ActiveJob::Arguments::TYPE_WHITELIST+,
+  # arrays, hashes, types that mixin +GlobalID::Identification+ (e.g.
+  # ActiveRecord models), Arrays and hashes must have serializable values,
+  # and hashes must also have string and symbol keys This error will also
+  # be raised when GlobalID fails to create an identifier for the argument as
+  # would happen if you tried to serialize an unpersisted ActiveRecord object.
+  class SerializationError < ArgumentError; end
```

Comment:
How about this?

> Raised when trying to serialize an unidentifiable GlobalID object (ex: unpersisted Active Record model).

What bothers me is that "GlobalID object" isn't entirely correct... but that phrasing does get the point across. Anything else ("objects that can be represented as GlobalIDs") seems overly verbose.